From 6890d88b7f50637cea838293388779c5f806d5e7 Mon Sep 17 00:00:00 2001
From: Reverend Homer <mk.43.ecko@gmail.com>
Date: Wed, 18 Mar 2015 23:31:41 +0300
Subject: [PATCH] Simple memory leak fix in playlist-spl.c

---
 src/playlist-spl.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/playlist-spl.c b/src/playlist-spl.c
index 2942b4c..5a6a3b0 100644
--- a/src/playlist-spl.c
+++ b/src/playlist-spl.c
@@ -615,6 +615,7 @@ static void spl_text_t_from_tracks(text_t** p,
     if(f != NULL) {
       append_text_t(&c, f);
       LIBMTP_PLST_DEBUG("track %d = %s (%u)\n", i+1, f, tracks[i]);
+      free(f);
     }
     else
       LIBMTP_ERROR("failed to find filepath for track=%d\n", tracks[i]);
-- 
2.7.4

